agatesql/table.py

```diff
@@ -194,19 +194,26 @@ def make_sql_table(table, table_name, dialect=None, db_schema=None, constraints=
         sql_column_kwargs = {}
 
         if constraints:
-            if isinstance(column.data_type, agate.Text) and dialect == 'mysql':
+            if isinstance(column.data_type, agate.Text) and dialect in ('ingres', 'mysql'):
                 length = table.aggregate(agate.MaxLength(column_name)) * decimal.Decimal(col_len_multiplier)
                 if length > 21844:
                     # @see https://dev.mysql.com/doc/refman/5.7/en/string-type-overview.html
                     sql_column_type = TEXT
+                elif dialect == 'ingres' and length > 16000:
+                    # @see https://docs.actian.com/actianx/11.2/index.html#page/SQLRef%2FCharacter_Data_Types.htm%23
+                    # Could be longer but ~16K is a reasonable cut off to switch to CLOB
```

Comment:
The preceding condition is "length > 21844" (which is 65,535 bytes divided by 3). The Actian docs mention 32,000 bytes, so not sure how you chose 16,000. I'll go with 10,666 (32,000 divided by 3) https://docs.actian.com/ingres/11.2/index.html#page/SQLRef/Character_Data_Types.htm

Reply:
Agreed, ~10K is a better safe/sane default. It was a while ago so like you, I cannot figure out how I chose ~16K either 😜

Thanks for taking the time to review and improve the original idea! Happy New Year 🎉